File: 프로그래머스/3/43238. 입국심사/입국심사.py
def solution(n, times):
    answer = 1e9
    times.sort()
    s, e = 0, n * times[-1]

    while s <= e:
        m = (s + e) // 2 # 전체 시간
        
        temp = 0
        for time in times:
            temp += m // time
            
        if temp < n:
            s = m + 1
        else:
            answer = m
            e = m - 1
        
    return answer

File: 프로그래머스/3/43238. 입국심사/입국심사.js
function solution(n, times) {
    var answer = 0;
    times.sort((a, b) => a-b);

    let [s, e] = [0, n*times[times.length-1]];
    
    while (s <= e) {
        const m = Math.floor((s+e)/2);
        
        let temp = 0;
        for ( const time of times ) {
            temp += Math.floor(m/time);
        }
        
        if (temp >= n) {
            answer = m;
            e = m - 1;
        } else {
            s = m + 1;
        }
        
    }
    
    return answer;
}